### PR TITLE
Finetune bicycle routing

### PR DESCRIPTION
**Pull request by leonardehrenfried**
Previously the local, regional and national cycle networks were heavily favoured. This works quite well when routing from town to town.

However, inside cities this would sometimes lead to very long detours as a route via the cycling network was selected instead.

In this PR I decreased the safety factor of the regional cycling networks a little and increased the safety of residential roads with a speed limit.

We have two exemplary test cases:

Nebringen to Herrenberg now produces this route (changed the coordinates slightly): https://gist.github.com/leonardehrenfried/00bc74682275a8b941b1623f383a8fb4

And the route along Erhardtstrasse is now routed like this: https://gist.github.com/leonardehrenfried/89dd1438714c3e95a63807908b2cb27a

Of course there will probably be other routes where this doesn't work so well, so my suggestion is to keep an eye on it and add more test cases as they come up.

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
Travis status because it still doesn't report: https://travis-ci.org/github/mfdz/OpenTripPlanner/builds/676179141

**Comment by leonardehrenfried** [comment]
https://travis-ci.org/github/mfdz/OpenTripPlanner/builds/676210373

File: pom.xml [hunk 1]
```diff
@@ -465,8 +465,10 @@
         </repository>
         <repository>
             <id>osgeo</id>
-            <name>Open Source Geospatial Foundation Repository</name>
-            <url>https://download.osgeo.org/webdav/geotools/</url>
+            <name>OSGeo Release Repository</name>
+            <url>https://repo.osgeo.org/repository/release/</url>
```
Comment (hbruch) [inline pom.xml:469]: I'm ok with this change, though it relates not originally with this issue?

File: src/main/java/org/opentripplanner/graph_builder/module/osm/GermanyWayPropertySetSource.java [hunk 2]
```diff
@@ -69,8 +69,8 @@ public void populateProperties(WayPropertySet props) {
 
         /** We assume highway/cycleway of a cycle network to be safer (for bicycle network relations, their network is copied to way in postLoad */
         props.setProperties("lcn=yes", StreetTraversalPermission.ALL, 0.7, 0.7, true); // local cycle network
-        props.setProperties("rcn=yes", StreetTraversalPermission.ALL, 0.5, 0.5, true); // regional cycle network
-        props.setProperties("ncn=yes", StreetTraversalPermission.ALL, 0.9, 0.9, true); // national cycle network
+        props.setProperties("rcn=yes", StreetTraversalPermission.ALL, 0.7, 0.7, true); // regional cycle network
```
Comment (hbruch) [inline src/main/java/org/opentripplanner/graph_builder/module/osm/GermanyWayPropertySetSource.java:72]: I'm ok with these. In the long run, these scaling factors should be evaluated in an OR-condition, currently they are multiplied each and every one, resulting in a very heavy preference, when all three networks run over the same way, which is not intended.

File: src/main/java/org/opentripplanner/routing/core/SimpleIntersectionTraversalCostModel.java [hunk 3]
```diff
@@ -30,20 +30,32 @@
...
             return 0;
         }
 
-        // Non-driving cases are much simpler. Handled generically in the base class.
-        if (!mode.isDriving()) {
-            return computeNonDrivingTraversalCost(v, from, to, fromSpeed, toSpeed);
+        if (mode.isDriving()) {
+            return computeDrivingTraversalCost(v, from, to, request);
+        }
+        else if(mode.isCycling()) {
```
Comment (hbruch) [inline src/main/java/org/opentripplanner/routing/core/SimpleIntersectionTraversalCostModel.java:50]: An alternative would be to leave SimpleIntersectionTraversalCostModel as is and have our changes in a new CostModel. That way, we'd minimize conflicts with others and allow people to switch the implementation.

File: pom.xml (hunk 1 above)
Comment (leonardehrenfried) [reply pom.xml:469]: No, it doesn't. geotools has moved to another repo: https://github.com/opentripplanner/OpenTripPlanner/issues/3034

**Pull request merged**